File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        int n = nums.size(); 

        for (int i = 0; i < n; i++){
            int find = target -  nums[i];
            if (numMap.count(find)){ //does this 'find' nubmer exist in the hashmap already?
                return {numMap[find], i}; //if it does, reutnr the index of where
            }
            numMap[nums[i]] = i; //If not, add the number at index i at the hashmap
        }

        return {}; //No solution found 
        
        
    }
};

File: Data Structures & Algorithms/single-number/submission-1.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:

        new = set() #creating unordered set

        for i in nums: #for every number in nums
            if i in new: # i exists in set? 
                new.remove(i)
            else: # i does not exist in set 
                new.add(i)

        return list(new)[0]